imap/http_dav.c

```diff
@@ -3816,7 +3816,7 @@ int parse_xml_body(struct transaction_t *txn, xmlNodePtr *root,
         return HTTP_BAD_REQUEST;
     }
     else {
-        xmlErrorPtr err = xmlCtxtGetLastError(txn->conn->xml);
+        const xmlError *err = xmlCtxtGetLastError(txn->conn->xml);
```

Comment:
@ksmurchison can you have a look at this one please, I'm not familiar with the XML library.

In particular, a few lines later we save `err->message` in `txn->error.desc`.  That might not be safe if we don't have lifetime ownership of the `err` object, which the constness might imply.

Since we aren't getting this warning/error outselves, perhaps this is an API change in the XML library that we haven't stumbled upon yet.

Reply:
As an information, the Arch distribution uses [libxml2 v2.12.3][1]. I see the changes at [libxml2 GitLab repository][2] which related to [this issue][3].  This is what maintainer's say:

> This is a slight break of the API, but users really shouldn't modify the global error struct. The goal is to make xmlLastError use static buffers for its strings eventually. This should warn people if they're abusing the struct.

[1]: https://gitlab.gnome.org/GNOME/libxml2/-/commits/v2.12.3
[2]: https://gitlab.gnome.org/GNOME/libxml2/-/commit/61034116d0a3c8b295c6137956adc3ae55720711
[3]: https://gitlab.gnome.org/GNOME/libxml2/-/commit/45470611b047db78106dcb2fdbd4164163c15ab7

Reply:
> Adding const seems correct as that is what the API specifies. Since we're getting the error from the XML parser context, and there is a xmlCtxtResetLastError() function, I assume its safe to reference the error until we free the parser context.

Alright, thanks for the clarification. Is there anything I can do?